### PR TITLE
docs(adapter): mention new types

### LINKED ISSUE #1881: Mention the new Adapters interface in the TS docs
## Summary 💭

Explain briefly how to utilize the new `Adapter` interface for better DX, either

- For TS users: https://twitter.com/nextauthjs/status/1387529559822766081
- For JS (JSDoc) users: https://twitter.com/nextauthjs/status/1387529925352202241

Thinking of a new section under Module augmentation at https://next-auth.js.org/getting-started/typescript

## Description 📓

We recently made it much easier for people to contribute to our new https://github.com/nextauthjs/adapters repository by adding their own custom adapters. We will take care of the governing (release, build, test etc.), so you just bring us the code!

## Contributing 🙌🏽

Up for grab! Good first issue :slightly_smiling_face:

### PR DESCRIPTION
**Pull request by 0ubbe**
## Reasoning 💡

Following #1881 mention the new **Adapter** type utilities in our documentation in:
- [Typescript](https://next-auth.js.org/getting-started/typescript)
- [Adapters](https://next-auth.js.org/schemas/adapters)

Specify the two ways to use them (in **TS** or through **JSDoc**).

## Checklist 🧢

- [x] Documentation
- [ ] ~Tests~
- [ ] ~Ready to be merged~
      

## Affected issues 🎟

Closes #1881

### CODE REVIEW

**Review by ndom91 — APPROVED** [review]
LGTM 👍

File: www/docs/getting-started/typescript.md [hunk 1]
```diff
@@ -100,6 +100,36 @@ declare module "next-auth/jwt" {
...
+`Adapter` can be [configured through generics](https://github.com/nextauthjs/next-auth/blob/main/types/adapters.d.ts#L120-L131) to suit the needs of your new adapter:
+
+```ts
+import type { Adapter } from "next-auth/adapters"
+
+interface AdapterConfig {
+  foo: string
+}
+
+const MyAdapter: Adapter<AdapterConfig> = () => {
```
Comment (balazsorban44) [inline www/docs/getting-started/typescript.md:130]: Per #1914 we should refer to the first generic as client, and all additional options should be passed in the second generic

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
The adapters section slipped under the Module Augmentation part, but I think it should be top level. The useful links section is supposed to be under Module Augmentation also (it is now just the Adapters section came accidentally before it).

Maybe a few words about JSDoc and how that can be utilized for IntelliSense even without TS?

Also, should we show all the available generics of the  Adapter rather than just the first one?

**Comment by lluia** [comment]
@balazsorban44 I addressed your suggestions 😄

File: www/docs/getting-started/typescript.md [hunk 2]
```diff
@@ -14,6 +14,43 @@ The types at [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyType
...
+      }
+    },
+  }
+}
+```
+
+When writing your own custom adapter in plain JavaScript, note that you can use **JSDoc** to get helpful editor hints and auto-completion like so:
+
+```js
+/** @type { import("next-auth/adapters").Adapters } */
```
Comment (balazsorban44) [inline www/docs/getting-started/typescript.md:38]: `Adapter`! 😁

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
thanks @lluia, found two small typos

File: www/docs/getting-started/typescript.md [hunk 3]
```diff
@@ -14,6 +14,43 @@ The types at [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyType
...
+      return {
+        // your adapter methods here
+      }
+    },
+  }
+}
+```
+
+:::note
+This will work in code editors with a strong Typescript integration like VSCode or WebStorm. It might not work if you're using more lightweight editors like VIM or Atom.
```
Comment (balazsorban44) [inline www/docs/getting-started/typescript.md:51]: TypeScript

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: www/docs/getting-started/typescript.md (hunk 2 above)
Comment (balazsorban44) [reply www/docs/getting-started/typescript.md:38]: Still seeing `import("next-auth/adapters").Adapters` The interface is called `Adapter`

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
@lluia I think you misunderstood my comment. :smile: I could have been more verbose...

**Comment by lluia** [comment]
@balazsorban44 hopefully now is right 🙃

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**